flink-runtime/src/main/java/org/apache/flink/runtime/util/FailJobExceptionHandler.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.util;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.function.Consumer;
+
+/**
+ * Handler for uncaught exceptions that will log the exception and fail the job afterwards.
+ *
+ * <p>This guarantees that critical exceptions are not accidentally lost and leave the system
+ * running in an inconsistent state.
+ */
+public final class FailJobExceptionHandler implements Thread.UncaughtExceptionHandler {
```

Comment:
Name is a leftover of the time it actually called the `context.failjob()` now that we changed it to `Cosumer<Throwable>` I agree it is misleading and BTW it is a class not a lambda because the is already exception handlers in the code base so I wanted to be consistent.

Reply:
If you prefer, as `FailJobExceptionHandler` is only a `Consumer<Throwable>` in the end, I can make this class a lambda `(Thread t, Throwable e) -> {context.failjob()}`. WDYT ?

Reply:
As it was what you suggested in the first place, I did the change above